File: day01/report_repair.py
#!/usr/bin/env python3
from sys import argv
from itertools import combinations

filename = argv[1]
with open(filename) as f:
    entries = [int(line) for line in f]

for pair in combinations(entries, 2):
    if sum(pair) == 2020:
        res = pair[0] * pair[1]
        break
print(res)

File: day01/report_repair.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.32);
use experimental qw(signatures);
use Algorithm::Combinatorics qw(combinations);
use List::Util qw(sum product);

my @entries = <>;
chomp @entries;

say 'Part 1: ', find_result(\@entries, 2);
say 'Part 2: ', find_result(\@entries, 3);

sub find_result($entries, $k) {
    my $iter = combinations($entries, $k);
    while (my $p = $iter->next) {
        return product($p->@*) if sum($p->@*) == 2020;
    }
}
